docs/oasis-network-primer/staking_rewards/Chart.tsx

```diff
@@ -0,0 +1,139 @@
+  return scaleRewards
+    .map(scale => {
+      const rewardPerEpoch = scale / rewardAmountDenominator;
+      const yearlyCompoundedRate = (1 + rewardPerEpoch)**estimatedEpochsPerYear;
+      return yearlyCompoundedRate - 1;
+    });
+};
+
+
+const StakingRewardsChart = () => {
+  const chart: PlotlyDataLayoutConfig = {
+    data: [
+      {
+        x: estimateEpochDates(data.map(d => d.untilEpoch)),
+        y: estimateAPY(data.map(d => d.scale)),
+        mode: 'lines',
+        name: 'Annualized rewards',
+        marker: {color: '#4285F4'},
+        text: data.map(d => `<i>Reward per epoch: ${d.scale * 100 / rewardAmountDenominator}%<br>Epoch: ${d.untilEpoch}</i>`),
+        hoverinfo: 'all',
+      },
+      {
+        x: [new Date().toISOString(), new Date().toISOString()],
+        y: [0, 0.205],
+        mode: 'lines',
+        name: 'Today',
+        marker: {color: 'gray'},
+        hoverinfo: 'none',
+        showlegend: false,
+      },
+    ],
+
+    layout: {
+      title: {
+        text: 'Rewards Schedule, approximated from block height of network',
+        font: {
+          size: 20,
+        },
+        yref: 'container',
+        y: 0.95,
+      },
+      xaxis: {
+        tickangle: -45,
+      },
+      yaxis: {
+        title: 'Rewards',
+        fixedrange: true,
+        tickformat: '0%',
+        hoverformat: '.3%',
+        dtick: 0.02,
+        rangemode: 'nonnegative',
+      },
+      margin: {
+        r: 0,
+      },
+      showlegend: false,
+      hoverlabel: {
+        namelength: 30,
+      },
+      hovermode: 'x unified',
+    },
+
+    config: {
+    },
+  };
+
+  // To generate fallback.svg:
+  // chart.data = chart.data.filter(d => d.name !== 'Today');
+  // chart.layout!.xaxis!.tickangle = -90;
+  // chart.layout!.xaxis!.nticks = 30;
+  // const PlotlyBasic: typeof import('plotly.js-basic-dist') = require('plotly.js-basic-dist');
+  // PlotlyBasic.downloadImage(chart, { filename: 'fallback', format:'svg', width: 960, height: 600 });
+
+  return <PlotlyChart label="Rewards Schedule, approximated from block height of network" chart={chart} fallbackSvg={FallbackSvg} />;
+};
+
+export default StakingRewardsChart;
+
+function testEstimatedEpochDates() {
+  const diffHoursFormatter = new Intl.NumberFormat(undefined, {
+    style: 'unit',
+    unit: 'hour',
+    unitDisplay: 'long',
+    signDisplay: 'always',
+    maximumFractionDigits: 1,
+  });
+
+  console.info('DEV: estimatedEpochsPerHour', estimatedEpochsPerHour);
+  console.info(
+    'DEV: diff between estimateEpochDates and known dates',
+    Object.entries(knownOffsets).map(([epochName, {epoch, date}]) => {
+      const estimatedTime = new Date(estimateEpochDates([epoch])[0]).getTime();
+      const indexedTime = new Date(date).getTime();
+      const diffHours = (indexedTime - estimatedTime) / 1000 / 60 / 60;
+      return `at ${epochName} (${epoch}): ${diffHoursFormatter.format(diffHours)}`;
+    })
+  );
```

Comment:
estimatedEpochsPerHour 1.003339695445333

diff between estimateEpochDates and known dates:
  - at beta (epoch 0): +14.1 hours
  - at mainnet (1170): +0 hours
  - at block4M (6664): +9 hours
  - at block5M (8328): +5.8 hours
  - at block6M (9992): -1.3 hours
  - at block7M (11656): +2 hours
  - at block8M (13320): +17.3 hours
  - at block9M (14987): +5.4 hours
  - at epoch 15835: +0 hours

Reply:
Thanks for computing this @lukaw3d! Very interesting data.